substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/debugentry/MethodEntry.java

```diff
@@ -72,26 +74,51 @@ public String getParamName(int idx) {
         return paramNames[idx];
     }
 
-    public boolean match(String methodName, String paramSignature, String returnTypeName) {
-        if (!methodName.equals(this.memberName)) {
-            return false;
+    public int compareTo(String methodName, String paramSignature, String returnTypeName) {
+        if (!memberName.equals(methodName)) {
```

Comment:
I would build this up as
```java
int memberNameCompare = memberName.compareTo(methodName);
if (memberNameCompare != 0) {
    return memberNameCompare;
}
int valueTypeCompare = valueType.getTypeName().compareTo(returnTypeName);
if (valueTypeCompare != 0) {
... etc etc ...
```

Reply:
Refactored, but as:
```
int result = memberName.compareTo(methodName);
if (result == 0) {
    result = valueTypeCompare = valueType.getTypeName().compareTo(returnTypeName);
}
if (result == 0) {
    ... etc etc ...
return result;
```

instead of as suggested. I find this equally readable while a bit more compact.

Reply:
Hmm, so if the first assignment to `result` returns a value `!=` `0` you will have have a couple more `if`s to go though before finally returning that value assigned in the beginning although it was clear from the beginning that this is the value that was going to be returned? :thinking:

Reply:
Of course the compiled version is likely to have optimized early exits. But while it runs in the interpreter it does exactly what you ask it to do. Also I find it easier to reason about code where locals are all effectively final (which is not the case in your variant)

Anyway, if you prefer this style I won't prevent you from merging it like that. I'm just providing my 2-cents.